<table>
    <tr>
        <td>
            <h1>Group 7</h1>
        </td>
        <td>
            <img src="https://colab.research.google.com/img/colab_favicon_256px.png" width="120"/><br>
            <a href="https://colab.research.google.com/github/mdazharuddin1011999/GAN/blob/master/5_pix2pix/1_data_process_colab.ipynb">Open with Colab</a>
        </td>
    </tr>
</table>

#### Download dataset from http://efrosgans.eecs.berkeley.edu/pix2pix/datasets/maps.tar.gz

In [1]:
!wget http://efrosgans.eecs.berkeley.edu/pix2pix/datasets/maps.tar.gz

'wget' is not recognized as an internal or external command,
operable program or batch file.


#### Extract the images from the downloaded tar file

In [ ]:
!tar -xf maps.tar.gz

In [ ]:
!ls

maps  maps.tar.gz  sample_data


#### Our images are inside the maps directory
**maps directory contains two sub-directories: train, val**  
**We have used train images only**

In [ ]:
from os import listdir

#### Get the name of all images inside train directory

In [ ]:
files = listdir("maps/train")

In [ ]:
from tensorflow.keras.preprocessing.image import load_img
from tensorflow.keras.preprocessing.image import img_to_array

In [ ]:
from tqdm import tqdm

In [ ]:
src_image=[]
tar_image=[]

**Each train image contains both source and target (i.e satellite view and google maps view)**  
**Split and store them in src_image and tar_image list respectively**

In [ ]:
for i in tqdm(range(len(files))):
    pixels = load_img("maps/train/"+files[i], target_size=(256, 512))
    pixels = img_to_array(pixels)
    sat_image, map_image = pixels[:, :256], pixels[:, 256:]
    src_image.append(sat_image)
    tar_image.append(map_image)

100%|██████████| 1096/1096 [00:13<00:00, 80.96it/s]


In [ ]:
from numpy import asarray

src_image = asarray(src_image, dtype="uint8")
tar_image = asarray(tar_image, dtype="uint8")

In [ ]:
print(src_image.shape, tar_image.shape)

(1096, 256, 256, 3) (1096, 256, 256, 3)


**So there are total 1096 images**  
**Save both src_images (source) and tar_images (target) lists in compressed form**

In [ ]:
from numpy import savez_compressed

savez_compressed("maps.npz", src_image, tar_image)

In [ ]:
!ls -sh

total 452M
4.0K maps  213M maps.npz  239M maps.tar.gz  4.0K sample_data


**Export the compressed file to Google Drive to be used later**

In [1]:
from google.colab import drive
drive.mount("/content/gdrive")

Mounted at /content/gdrive


In [2]:
!mkdir "gdrive/MyDrive/7th Sem Project"

mkdir: cannot create directory ‘gdrive/MyDrive/7th Sem Project’: File exists


In [ ]:
!cp maps.npz "gdrive/MyDrive/7th Sem Project/maps.npz"

In [ ]:
!ls "gdrive/My Drive/7th Sem Project"

maps.npz
